src/tests.rs

```diff
@@ -17,6 +17,11 @@ pub(crate) fn setup<'a>(medium: Medium) -> (Interface, SocketSet<'a>, TestingDev
         ])),
     });
 
+    #[cfg(feature = "medium-ieee802154")]
```

Comment:
This (and the changes to the unit tests) appear unrelated to the larger change. Can you break these out into a separate commit (and maybe separate PR)?

Reply:
Sorry for that! I made a separate commit for it and not a separate PR since this PR would fail without it.